app/src/main/java/com/getsensibill/spendmanager/demo/subclass/SubclassJavaActivity.java

```diff
@@ -0,0 +1,149 @@
+     * you can override this and provide a different [NavigationIntent]. Alternatively,
+     * you can also apply this as an argument when overriding [createWebUiFragment].
+     */
+    @Override
+    protected @NotNull NavigationIntent getWebInitialNavigation() {
+        return NavigationIntent.DASHBOARD.INSTANCE;
+    }
+
+    /**
+     * Not required override.
+     *
+     * Used to create the [WebUiFragment] held in the Activity. Common reason for overriding
+     * this would be if there are arguments that are needed to be supplied to the fragment.
+     */
+    @Override
+    protected @NotNull WebUiFragment createWebUiFragment() {
+        WebUiFragment fragment = super.createWebUiFragment();
+        Bundle bundle = new Bundle();
+
+        // Pass in a navigation override. Defaults as .DASHBOARD
+        bundle.putParcelable(WebUiFragment.ARG_NAVIGATION_OVERRIDE, NavigationIntent.DASHBOARD.INSTANCE);
+
+        // Pass in a custom Theme override. Defaults to null
+        ProgrammaticTheme theme = new ProgrammaticTheme(new Brand());
+        bundle.putParcelable(WebUiFragment.ARG_PROGRAMMATIC_THEME_OVERRIDE, theme);
+
+        fragment.setArguments(bundle);
+        return fragment;
+    }
+
+    /**
+     * Not required override.
+     *
+     * The shouldNavigateBack from the [webFragment] will inform you if your activity
+     * should handle the back pressed normally. Also note, that if you need direct access to
+     * activity.onBackPressed, we have [activityOnBackPressed] convenience method you can call.
+     */
+    @Override
+    public void onBackPressed() {
+        final WebUiFragment webFragment = getWebFragment();
+        final boolean shouldNavigateBack = webFragment == null || webFragment.shouldNavigateBack();
+        if (shouldNavigateBack) {
+            finish();
+        }
+    }
+
+    /**
+     * Not required override.
+     *
+     * This will be called when ending the WebUi flow, and will give you a [UiFinishReason]
+     * which can be used to identify why the flow is ending. Default behaviour is to try and
+     * reload the Spend Manager. If you do not want it to attempt an immediate reload of
+     * Spend Manager, you can override this call.
+     */
+    @Override
+    public void onRequestFinish(@NotNull UiFinishReason reason) {
+        super.onRequestFinish(reason);
+        Timber.i(reason.name());
+    }
+
+    /**
+     * Not required override.
+     *
+     * By overriding [onDisplayNetworkError], handle when the network error fragment needs to be
+     * navigated to. If you do not override it, then by default Spend Manager will handle when and
+     * how to display the [WebUiNetworkErrorFragment].
+     */
+    @Override
+    public void onDisplayNetworkError(boolean networkNotAvailable) {
+        super.onDisplayNetworkError(networkNotAvailable);
+        Timber.i(networkNotAvailable ? "true" : "false");
```

Comment:
It's been so long since we haven't had the convenience of Kotlin haha.
But just to check, can we use `String.valueOf(nna)`?

Reply:
looks like it. updated